### PR TITLE
[Communication] - PhoneNumbers - Update to GA swagger

### PR DESCRIPTION
**Pull request by pavelprystinka**
API review: https://apiview.dev/Assemblies/Review/b32afd45c457422992f506154f376d68
Swagger update PR: https://github.com/Azure/azure-rest-api-specs/pull/12446/

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Common/tests/CommunicationRecordedTestSanitizer.cs [hunk 1]
```diff
@@ -13,6 +14,7 @@ public class CommunicationRecordedTestSanitizer : RecordedTestSanitizer
         private static readonly Regex _azureResourceRegEx = new Regex(@"[^/]+?(?=(.communication.azure))", RegexOptions.Compiled);
         private static readonly Regex _identityInRouteRegEx = new Regex(@"(?<=identities/)([^/]+)", RegexOptions.Compiled);
         private static readonly Regex _phoneNumberRegEx = new Regex(@"[\\+]?[0-9]{11,15}", RegexOptions.Compiled);
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Common/tests/CommunicationRecordedTestSanitizer.cs:16]: your baseline seems off of master. The phone number sanitizations are moved to the PhoneNumbers.Tests project

File: sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - Added PhoneNumberAdministrationClient (originally was part of the Azure.Communication.Administration package).
 - Added support to create PhoneNumberAdministrationClient with AzureKeyCredential.
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md:7]: The client is completely changed, please check with @minnieliu and make sure we align the changelogs

File: sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Azure.Communication.PhoneNumbers.Models;
+using Azure.Core;
+
+#nullable enable
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs:11]: now that `<Nullable>enable</Nullable>` is removed from the csproj file, we should not explicitly disable here again

File: sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                using var document = JsonDocument.Parse(content);
+
+                foreach (var property in document.RootElement.EnumerateObject())
+                {
+                    if (property.NameEquals("error"))
+                    {
+                        var communicationError = CommunicationError.DeserializeCommunicationError(property.Value);
+                        errorCode = communicationError.Code;
+                        message = communicationError.Message;
+                        additionalInfo = new Dictionary<string, string>() { { "target", communicationError.Target } };
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs:54]: nit: `additionalInfo = new Dictionary<string, string>() { ["target"] = communicationError.Target };

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+        //[TestCase("geographic", "user", "inbound", "none")]
+        //[TestCase("geographic", "application", "outbound", "none")]
+        //[TestCase("geographic", "application", "inbound+outbound", "none")]
+        //[TestCase("geographic", "user", "inbound", "inbound")]
+        //[TestCase("geographic", "application", "outbound", "outbound")]
+        //[TestCase("geographic", "application", "inbound+outbound", "inbound+outbound")]
+        //[TestCase("geographic", "user", "none", "inbound")]
+        //[TestCase("geographic", "application", "none", "outbound")]
+        //[TestCase("geographic", "application", "none", "inbound+outbound")]
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs:76]: is this intentional?

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+            Assert.AreEqual(searchedNumber, phoneNumber.PhoneNumber);
+
+            Assert.AreEqual(phoneType, phoneNumber.PhoneNumberType.ToString());
+            Assert.AreEqual(assignmentType, phoneNumber.AssignmentType.ToString());
+            Assert.AreEqual(calling, phoneNumber.Capabilities.Calling.ToString());
+            Assert.AreEqual(sms, phoneNumber.Capabilities.Sms.ToString());
+
+            // TODO:
+            // Assert.AreEqual(sms, phoneNumber.Cost);
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs:135]: Is this still supposed to be here?

File: sdk/communication/Azure.Communication.Common/tests/CommunicationRecordedTestSanitizer.cs (hunk 1 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Common/tests/CommunicationRecordedTestSanitizer.cs]: I have it rebased.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs (hunk 3 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs]: Removed

File: sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs (hunk 4 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/ClientDiagnostics.cs]: Applied

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs (hunk 5 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs]: Removed

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs (hunk 6 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumbersClientLiveTests.cs]: Removed

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 7]
```diff
@@ -93,11 +115,115 @@ private PhoneNumbersClient(string endpoint, TokenCredential tokenCredential, Pho
...
 
         #endregion
 
+        /// <summary> Releases an acquired phone number. </summary>
+        /// <param name="phoneNumber"> Phone number to be released, e.g. +11234567890. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="phoneNumber"/> is null. </exception>
+        public virtual async Task<ReleasePhoneNumberOperation> StartReleasePhoneNumberAsync(string phoneNumber, CancellationToken cancellationToken = default)
+        {
+            if (phoneNumber == null)
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:129]: Let's use `Argument.AssertNotNull`

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 8]
```diff
@@ -78,8 +80,28 @@ public PhoneNumbersClient(Uri endpoint, TokenCredential tokenCredential, PhoneNu
 
         #endregion
 
+        #region internal constructors
+        /// <summary> Initializes a new instance of PhoneNumbersClient. </summary>
+        /// <param name="clientDiagnostics"> The handler for diagnostic messaging in the client. </param>
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="endpoint"> The communication resource, for example https://resourcename.communication.azure.com. </param>
+        /// <param name="apiVersion"> Api Version. </param>
+        internal PhoneNumbersClient(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, string endpoint, string apiVersion = "2021-03-07")
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:89]: why do we need this constructor?

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 9]
```diff
@@ -93,11 +115,115 @@ private PhoneNumbersClient(string endpoint, TokenCredential tokenCredential, Pho
...
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="phoneNumber"/> is null. </exception>
+        public virtual ReleasePhoneNumberOperation StartReleasePhoneNumber(string phoneNumber, CancellationToken cancellationToken = default)
+        {
+            if (phoneNumber == null)
+            {
+                throw new ArgumentNullException(nameof(phoneNumber));
+            }
+
+            using var scope = _clientDiagnostics.CreateScope("PhoneNumbersClient.StartReleasePhoneNumber");
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:159]: `$"{nameof(PhoneNumbersClient)}.{nameof(StartReleasePhoneNumber)}"`. Same for others too

File: sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md [hunk 10]
```diff
@@ -6,10 +6,14 @@
...
 - Added support to create PhoneNumberAdministrationClient with TokenCredential
+- Added support for long running operations
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md:9]: "." at the end of line. Please update the other missing ones too.

File: sdk/communication/Azure.Communication.PhoneNumbers/README.md [hunk 11]
```diff
@@ -42,28 +42,28 @@ All geographic phone plans within the same country are grouped into a phone plan
...
-var client = new PhoneNumberAdministrationClient(new Uri(endpoint), tokenCredential);
+client = new PhoneNumbersClient(endpoint, tokenCredential);
 ```
 
-### Reserving and acquiring numbers
+### Phone Number Types overview
+Phone numbers come in two types: Geographic and Toll-Free. Geographic phone plans are phone plans associated with a location, whose phone numbers' area codes are associated with the area code of a geographic location. Toll-Free phone plans are phone plans not associated location. For example, in the US, toll-free numbers can come with area codes such as 800 or 888.
 
-Phone numbers reservation can be performed through the reservation creation API by providing a phone plan id, an area code and quantity of phone numbers. The provided quantity of phone numbers will be reserved for ten minutes. This reservation of phone numbers can either be cancelled or purchased. If the reservation is cancelled, then the phone numbers will become available to others. If the reservation is purchased, then the phone numbers are acquired for the Azure resources.
+### Searching and Purchasing and Releasing numbers
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/README.md:64]: Searching, Purchasing and Releasing Phone Numbers

File: sdk/communication/Azure.Communication.PhoneNumbers/README.md [hunk 12]
```diff
@@ -42,28 +42,28 @@ All geographic phone plans within the same country are grouped into a phone plan
...
+```C# Snippet:CreatePhoneNumbersClientWithTokenCredential
+// Get an endpoint to our Azure Communication resource.
+var endpoint = new Uri("<endpoint_url>");
 TokenCredential tokenCredential = new DefaultAzureCredential();
-var client = new PhoneNumberAdministrationClient(new Uri(endpoint), tokenCredential);
+client = new PhoneNumbersClient(endpoint, tokenCredential);
 ```
 
-### Reserving and acquiring numbers
+### Phone Number Types overview
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/README.md:61]: Phone number types overview

File: sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumberSearchRequest.cs [hunk 13]
```diff
@@ -1,9 +1,10 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core;
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumberSearchRequest.cs:4]: using probably not needed

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 14]
```diff
@@ -0,0 +1,458 @@
...
+using Azure.Communication.Pipeline;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.PhoneNumbers
+{
+    /// <summary>
+    /// The Azure Communication Services phone numbers client.
+    /// </summary>
+    [CodeGenSuppress("GetByNumberAsync", typeof(string), typeof(CancellationToken))]
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:17]: Now that this is not generated, we probably do not need these `CodeGenSupress` series, right?

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 15]
```diff
@@ -0,0 +1,458 @@
...
+            _clientDiagnostics = clientDiagnostics;
+            _pipeline = pipeline;
+        }
+
+        #endregion internal constructors
+
+        #region private constructors
+
+        /// <summary> Initializes a new instance of PhoneNumbersClient for mocking. </summary>
+        protected PhoneNumbersClient()
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:101]: let's move this ctor to after the private constructors region

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 8 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:89]: It is in use by other constructor.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 9 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:159]: Done

File: sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md (hunk 10 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md:9]: Done.

File: sdk/communication/Azure.Communication.PhoneNumbers/README.md (hunk 11 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/README.md:64]: Done

File: sdk/communication/Azure.Communication.PhoneNumbers/README.md (hunk 12 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/README.md:61]: Done

File: sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumberSearchRequest.cs (hunk 13 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumberSearchRequest.cs:4]: Removed

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 15 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:101]: Done

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 7 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:129]: Done.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 14 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:17]: The flag public-clients: true is still there to generate LROs.
We just suppress all not needed generated code and manually override the rest.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 8 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:89]: It is used only by last private ctor, let's just move it there and delete this extra one.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 14 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:17]: Did we not agree to make this internal and write the public client manually?

Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs]: Made generated client internal and removed CodeGenSuppress

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 8 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs]: Moved

**Review by RezaJooyandeh — APPROVED** [review]
I have one comment about the changelog. But let's do that in a separate PR

**Pull request merged**